velox/expression/Expr.cpp

```diff
@@ -639,7 +639,11 @@ void Expr::addNulls(
         SelectivityVector::empty(), type(), context->pool(), result);
   }
   if ((*result)->size() < rows.end()) {
-    (*result)->resize(rows.end());
+    // Not all Vectors support resize.  Since we only want to append nulls,
```

Comment:
I wonder if it make sense to fold this logic into BaseVector::addNulls?

Reply:
Just to clarify a bit, there is a BaseVector::mayAddNulls() method which returns true if the vector supports adding nulls. Wouldn't it make sense to make it so that if mayAddNulls() returns true, then addNulls() succeeds and doesn't require the caller to call resize()?

Reply:
I attempted to move this into addNulls, it seems dangerous though.

Here (and AFAICT only here) rows may be larger than result, but not all of the values that need to be appended may be null.  Even though these new non-null values aren't defined this is ok because due to peeling those non-null values won't actually be looked at.  In the general case when BaseVector.addNulls is called though adding undefined non-null values seems really dangerous.

I also considered making those values null, but again, this seems dangerous because we're adding nulls the caller wasn't necessarily expecting.  I think keeping it here is the safer choice.

Reply:
I think mayAddNulls() is poorly named, it's only called in 4 places in the code base, and 2 of those are in clearNulls() which doesn't call addNulls() at all.  I've renamed it to isNullsWritable() which I think better captures what the function is expressing.  This also I think resolves the confusion here a bit, while mayAddNulls makes it sound like it addNulls won't throw, isNullsWritable() doesn't give that impression.